src/Entity/Membership.php

```diff
@@ -298,4 +298,13 @@ public function setWorkflowId($workflow) {
     return $this;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public function getProviderPlugin() {
+    /** @var \Drupal\membership\Plugin\MembershipProviderManager $manager */
+    $manager = \Drupal::service('plugin.manager.membership_provider.processor');
```

Comment:
Not that I'm aware. Core's Node entity implementation, for instance, uses the `\Drupal` global class for retrieving services.